### PR TITLE
feat(DataTable): multi page select for table and small fix for shift select

### LINKED ISSUE #135: Table: support multi page selection
## Description

Add an option to choose which behaviour for the table selection
1. persist the selected rows in the table, and keep the selection even after the page/sort change, but reset it after search change.
2. reset the selection by sort/page/search change.

### PR DESCRIPTION
**Pull request by tewshi**
Closes #135

### CODE REVIEW

File: src/components/tables/DataTable.vue [hunk 1]
```diff
@@ -139,6 +139,7 @@ export interface Props {
    * When true, changing the items per page setting in one table will affect other tables.
    */
   globalItemsPerPage?: boolean;
+  singlePageSelect?: boolean;
```
Comment (kelsos) [inline src/components/tables/DataTable.vue:142]: wouldn't applying this basically change the default behavior on all of our tables?

Wouldn't be better to change it to multi-page select and keep the single page select as the default behavior?

File: src/components/forms/checkbox/Checkbox.vue [hunk 2]
```diff
@@ -72,6 +74,12 @@ const { hasError, hasSuccess } = useFormTextDetail(
   errorMessages,
   successMessages,
 );
+
+watch(value, (val) => {
+  const input = get(el);
+  if (input && input.checked !== val)
+    input.checked = val;
+});
```
[the remark this answers is not in the document]
Comment (tewshi) [reply src/components/forms/checkbox/Checkbox.vue:82]: when moving between pages, there's a bug where if you clicked the 'select-all' checkbox on a previous page, (with multi-page select feature on), then the checkbox doesn't behave as expected when you click on it... the input still maintains it's checked state (internally).

File: src/components/tables/DataTable.vue (hunk 1 above)
Comment (tewshi) [reply src/components/tables/DataTable.vue]: okay, will update...

I once was on a page where the table selection span multiple pages

Comment (kelsos) [reply src/components/tables/DataTable.vue]: Yes that was an exception though, I think most of the pages we have now are single by default (at least the ones with external pagination)

**Pull request merged**